pkg/reconcile/pipeline/handler/collect/impl.go

```diff
@@ -131,6 +132,27 @@ func ProvisionedService(ctx pipeline.Context) {
 	}
 }
 
+func DirectSecretReference(ctx pipeline.Context) {
+	services, err := ctx.Services()
+	if err != nil {
+		requestRetry(ctx, ErrorReadingSecret, err)
```

Comment:
The error might not be related to reading a secret in this particular case.

Reply:
Fixed. cf2da0c816f48b61546bb862397dace9fb581ec6